### PR TITLE
dbld: bump versioning.h in prepare-release

### PR DESCRIPTION
**Pull request by alltilla**
Note: it is enough to merge this PR after releasing 3.30.1, if we manually do these bumps this time for 3.30.1.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: dbld/prepare-release [hunk 1]
```diff
@@ -6,8 +6,12 @@ cd /source
 VERSION=$1
 
 OLD_VERSION=`cat VERSION`
-OLD_MAJOR=$(echo $OLD_VERSION | cut -d. -f1-2)
-NEW_MAJOR=$(echo $VERSION | cut -d. -f1-2)
+OLD_MAJOR=$(echo $OLD_VERSION | cut -d. -f1)
+OLD_MINOR=$(echo $OLD_VERSION | cut -d. -f2)
+OLD_PATCH=$(echo $OLD_VERSION | cut -d. -f3)
```
Comment (szemere) [inline dbld/prepare-release:11]: I like the symmetry in the code (also extracting PATCH from the version number), but unfortunately PATCH is never used. Could you give an example for that?

**Review by szemere — APPROVED** [review]

File: dbld/prepare-release (hunk 1 above)
Comment (alltilla) [reply dbld/prepare-release:11]: Yes, I extracted `PATCH` for the sake of completeness. I did not want to split the `VERSION` to 3 parts, then only provide 2 of them with variables.

**Comment by alltilla** [comment]
The logic here needs to be validated in the `dbld/rules release` step. I will do that this week.

**Comment by alltilla** [comment]
Ready for review.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by gaborznagy** [comment]
I've just noticed that `dh-systemd` is missing during docker image creation, which causes that [tarball](https://github.com/syslog-ng/syslog-ng/pull/3490/checks?check_run_id=1537112749) github actions job breaks.

Strangely, I don't find dh-systemd in testing(bullseye), but it's in Sid though.
https://packages.debian.org/search?searchon=names&keywords=dh-systemd

**Comment by alltilla** [comment]
@gaborznagy It is interesting. I think it is not related to this PR, but it definitely needs a fix, regardless.

**Comment by gaborznagy** [comment]
> @gaborznagy It is interesting. I think it is not related to this PR, but it definitely needs a fix, regardless.

`dh-systemd` seems to be a transitional package in `debian buster` [already](https://packages.debian.org/buster/dh-systemd), which means we can delete it.
I've checked `debhelper` package and it seems to contain the necessary binaries:
```
/# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@minke:/# 


root@minke:/# dpkg -L dh-systemd
/.
/usr
/usr/share
/usr/share/doc
/usr/share/doc/dh-systemd
/usr/share/doc/dh-systemd/changelog.gz
/usr/share/doc/dh-systemd/copyright


root@minke:/# dpkg -L debhelper | grep systemd
/usr/bin/dh_installsystemd
/usr/bin/dh_installsystemduser
/usr/bin/dh_systemd_enable
/usr/bin/dh_systemd_start

```

**Comment by gaborznagy** [comment]
I'll forgot to say, I'll open a PR once I've verified it works.

**Review by bazsi — DISMISSED** [review]
I like that we are moving the validation logic into a separate script, outside of the makefile.

**Review by szemere — APPROVED** [review]

**Comment by alltilla** [comment]
Rebased to master, to obtain the fix for "Binary packages".

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by szemere — APPROVED** [review]

**Review by bazsi — APPROVED** [review]

**Review by gaborznagy — APPROVED** [review]

**Pull request merged**